src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala

```diff
@@ -25,6 +90,62 @@ object Directives extends AuthDirectives with LeaderDirectives with AkkaDirectiv
     }
   }
 
+  private val marathonApiKeywords = Set("restart", "tasks", "versions")
+
+  //basically it's equivalent to path.takeWhile(s => !marathonApiKeywords.contains(s)) map createPathId
+  private[akkahttp] def findSomethingLikeAppId(remainingPath: Path, extractedAppName: Path = Path.Empty): Option[PathId] = {
```

Comment:
Hm, what does "something like" mean? "Like" can carry a lot of meaning.

Reply:
We're doing best effort here to find a wrong appid